File: app/src/main/java/ohi/andre/consolelauncher/commands/main/raw/termux.java
package ohi.andre.consolelauncher.commands.main.raw;

import android.content.Intent;
import android.os.Handler;
import android.os.Looper;

import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import ohi.andre.consolelauncher.R;
import ohi.andre.consolelauncher.UIManager;
import ohi.andre.consolelauncher.commands.CommandAbstraction;
import ohi.andre.consolelauncher.commands.ExecutePack;

public class termux implements CommandAbstraction {

    @Override
    public String exec(ExecutePack info) {
        String command = null;
        if (info.args != null && info.args.length > 0) {
            Object arg = info.get();
            if (arg != null) {
                command = arg.toString();
            }
        }

        openConsole(info, command);
        return null;
    }

    private void openConsole(ExecutePack info, String command) {
        Intent intent = new Intent(UIManager.ACTION_TERMUX_CONSOLE);
        intent.putExtra(UIManager.EXTRA_TERMUX_COMMAND, command);
        Handler handler = new Handler(Looper.getMainLooper());
        handler.post(() -> LocalBroadcastManager
                .getInstance(info.context.getApplicationContext())
                .sendBroadcast(intent));
    }

    @Override
    public int[] argType() {
        return new int[] {CommandAbstraction.PLAIN_TEXT};
    }

    @Override
    public int priority() {
        return 3;
    }

    @Override
    public int helpRes() {
        return R.string.help_termux;
    }

    @Override
    public String onArgNotFound(ExecutePack info, int index) {
        return info.context.getString(R.string.help_termux);
    }

    @Override
    public String onNotArgEnough(ExecutePack info, int nArgs) {
        openConsole(info, null);
        return null;
    }
}

File: app/src/main/java/ohi/andre/consolelauncher/commands/main/raw/termux.kt
@file:Suppress("DEPRECATION")

package ohi.andre.consolelauncher.commands.main.raw

import android.content.Intent
import android.os.Handler
import android.os.Looper
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import ohi.andre.consolelauncher.R
import ohi.andre.consolelauncher.UIManager
import ohi.andre.consolelauncher.commands.CommandAbstraction
import ohi.andre.consolelauncher.commands.ExecutePack

class termux : CommandAbstraction {
    override fun exec(pack: ExecutePack): String? {
        var command: String? = null
        val currentArgs = pack.args
        if (currentArgs != null && currentArgs.isNotEmpty()) {
            val arg = pack.get()
            if (arg != null) {
                command = arg.toString()
            }
        }

        openConsole(pack, command)
        return null
    }

    private fun openConsole(pack: ExecutePack, command: String?) {
        val intent = Intent(UIManager.ACTION_TERMUX_CONSOLE)
        intent.putExtra(UIManager.EXTRA_TERMUX_COMMAND, command)
        val handler = Handler(Looper.getMainLooper())
        handler.post {
            LocalBroadcastManager
                .getInstance(pack.context.applicationContext)
                .sendBroadcast(intent)
        }
    }

    override fun argType(): IntArray = intArrayOf(CommandAbstraction.PLAIN_TEXT)

    override fun priority(): Int = 3

    override fun helpRes(): Int = R.string.help_termux

    override fun onArgNotFound(pack: ExecutePack, indexNotFound: Int): String = pack.context.getString(R.string.help_termux)

    override fun onNotArgEnough(pack: ExecutePack, nArgs: Int): String? {
        openConsole(pack, null)
        return null
    }
}
